benchmarks/distributed/rpc/parameter_server/launcher.py

```diff
@@ -0,0 +1,320 @@
+        configurations.parameter_server_count
+    )
+    configured_args = ps_configurations["configurations"].values()
+    ps_trainer_count = configurations.trainer_count / configurations.parameter_server_count
+    rem = configurations.trainer_count % configurations.parameter_server_count
+    if parameter_server_rank - configurations.trainer_count < rem:
+        ps_trainer_count += 1
+    return rpc.remote(
+        name,
+        ps,
+        args=(
+            parameter_server_rank,
+            ps_trainer_count,
+            *configured_args,
+        ),
+    )
+
+
+def run_trainer(
+    configurations, trainer_configurations, model, data, rank, ps_rref
+):
+    trainer_class = get_benchmark_trainer_map()[str(trainer_configurations["trainer_class"])]
+    configured_args = trainer_configurations["configurations"].values()
+    trainer = trainer_class(
+        rank,
+        configurations.trainer_count,
+        ps_rref,
+        *configured_args
+    )
+    trainer.train(model, data)
+    metrics = trainer.get_metrics()
+    return [rank, metrics]
+
+
+def call_trainers(configurations, trainer_configurations, model, train_data, parameter_server_rrefs):
+    futs = []
+    for trainer_rank in range(0, configurations.trainer_count):
+        trainer_name = get_name(
+            trainer_rank,
+            configurations.trainer_count,
+            configurations.parameter_server_count
+        )
+        ps_rref = None
+        if parameter_server_rrefs:
+            ps_rank = get_parameter_server_rank(trainer_rank, configurations)
+            ps_rref = parameter_server_rrefs[ps_rank]
+        fut = rpc.rpc_async(
+            trainer_name,
+            run_trainer,
+            args=(
+                configurations,
+                trainer_configurations,
+                copy.deepcopy(model),
+                train_data[trainer_rank],
+                trainer_rank,
+                ps_rref,
+            ),
+            timeout=configurations.rpc_async_timeout
+        )
+        futs.append(fut)
+    return futs
+
+
+def benchmark_warmup(
+    configurations, trainer_configurations, ps_configurations, model, data, parameter_server_rrefs
+):
+    if configurations.parameter_server_count > 0:
+        ps = get_benchmark_ps_map()[str(ps_configurations["ps_class"])]
+    futs = call_trainers(configurations, trainer_configurations, model, data, parameter_server_rrefs)
+    for fut in futs:
+        fut.wait()
+    for ps_rref in parameter_server_rrefs.values():
+        rpc.rpc_sync(
+            ps_rref.owner(),
+            ps.reset_state,
+            args=(ps_rref,)
+        )
+    print("benchmark warmup done\n")
+
+
+def split_list(arr, n):
+    return [arr[i::n] for i in range(n)]
+
+
+def run_master(rank, model, data, configurations, trainer_configurations, ps_configurations, rpc_backend_options):
+    world_size = configurations.trainer_count + configurations.parameter_server_count + 1
+    rpc.init_rpc(
+        get_name(
```

Comment:
nit: will it be better if `get_name` just take the `configurations` as an argument and then retrieve whatever fields it needs? It will help make the call site less verbose.

Reply:
also in case there will be additional roles in the future, we don't need to update all call sites.

Reply:
okay updated the get_name method.